use class syntax in stream-view

batfink opened the pull request:
Since using util.inherits is now [discouraged](https://nodejs.org/docs/latest/api/util.html#util_util_inherits_constructor_superconstructor), I thought I’d rewrite the [stream-view example](https://github.com/koajs/examples/tree/master/stream-view) using the class syntax as part of my own learning process.

batfink commented:
- The Travis build fails because the version of node used to run the test is outdated. 
- Snyk complains about uglify.js

Fixing these issues is not within my jurisdiction I’m afraid.

dead-horse commented on .travis.yml:
```diff
@@ -1,4 +1,4 @@
 node_js:
-- "0.12"
+- "6.3"
```
```
node_js:
  "4"
  "6"
```

remove iojs and add node 4

batfink commented:
@dead-horse: Done. I also added 'use strict' to view.js (node 4 requirement) and removed the `--harmony` flag from the README since it’s no longer necessary. Rebased to a single commit.

dead-horse commented:
👍
